src/omnicore/encoding.cpp

```diff
@@ -22,57 +22,57 @@
  */
-bool OmniCore_Encode_ClassB(const std::string& senderAddress, const CPubKey& redeemingPubKey,
-        const std::vector<unsigned char>& vchPayload, std::vector<std::pair<CScript, int64_t> >& vecOutputs)
-{
-    unsigned int nRemainingBytes = vchPayload.size();
-    unsigned int nNextByte = 0;
-    unsigned char chSeqNum = 1;
-    std::string strObfuscatedHashes[1+MAX_SHA256_OBFUSCATION_TIMES];
-    PrepareObfuscatedHashes(senderAddress, MAX_SHA256_OBFUSCATION_TIMES, strObfuscatedHashes);
-    while (nRemainingBytes > 0) {
-        int nKeys = 1; // Assume one key of data, because we have data remaining
-        if (nRemainingBytes > (PACKET_SIZE - 1)) { nKeys += 1; } // ... or enough data to embed in 2 keys
-        std::vector<CPubKey> vKeys;
-        vKeys.push_back(redeemingPubKey); // Always include the redeeming pubkey
-        for (int i = 0; i < nKeys; i++) {
-            // Add up to 30 bytes of data
-            unsigned int nCurrentBytes = nRemainingBytes < (PACKET_SIZE - 1) ? nRemainingBytes: (PACKET_SIZE - 1);
-            std::vector<unsigned char> vchFakeKey;
-            vchFakeKey.insert(vchFakeKey.end(), chSeqNum); // Add sequence number
-            vchFakeKey.insert(vchFakeKey.end(), vchPayload.begin() + nNextByte,
-                                                vchPayload.begin() + nNextByte + nCurrentBytes);
-            vchFakeKey.resize(PACKET_SIZE); // Pad to 31 total bytes with zeros
-            nNextByte += nCurrentBytes;
-            nRemainingBytes -= nCurrentBytes;
-            std::vector<unsigned char> vchHash = ParseHex(strObfuscatedHashes[chSeqNum]);
-            for (size_t j = 0; j < PACKET_SIZE; j++) { // Xor in the obfuscation
-                vchFakeKey[j] = vchFakeKey[j] ^ vchHash[j];
-            }
-            vchFakeKey.insert(vchFakeKey.begin(), 0x02); // Prepend a public key prefix
-            vchFakeKey.resize(33);
-            CPubKey pubKey;
-            unsigned char chRandom = static_cast<unsigned char>(GetRand(256));
-            for (int j = 0; i < 256 ; j++) { // Fix ECDSA coodinate
-                vchFakeKey[32] = chRandom;
-                pubKey = CPubKey(vchFakeKey);
-                if (pubKey.IsFullyValid()) break;
-                ++chRandom; // ... but cycle no more than 256 times to find a valid point
-            }
-            vKeys.push_back(pubKey);
-            chSeqNum++;
-        }
-
-        // Push back a bare multisig output with obfuscated data
-        CScript scriptMultisigOut = GetScriptForMultisig(1, vKeys);
-        vecOutputs.push_back(std::make_pair(scriptMultisigOut, GetDustThreshold(scriptMultisigOut)));
-    }
-
-    // Add the Exodus marker output
-    CScript scriptExodusOutput = GetScriptForDestination(ExodusAddress().Get());
-    vecOutputs.push_back(std::make_pair(scriptExodusOutput, GetDustThreshold(scriptExodusOutput)));
-    return true;
-}
+// bool OmniCore_Encode_ClassB(const std::string& senderAddress, const CPubKey& redeemingPubKey,
```

Comment:
Just want to learn why it's removed

Reply:
In simple words: Class  B encoding is and old fashion way to embbeding data into the blockchain (using multisig addresses, when there was not OP_RETURN opcode), the outputs of those transactions include Exodus address, that implies "dust" . I removed this because i'm trying to use just Class C (using OP_RETURN ) for all our transactions. Actual Omni team wants to use Class D (similar to Class C but marker bytes "ol" rather than "omni" and integers in payloads are compressed with variable length encoding similar to LEB128).  If you want more info related : https://github.com/OmniLayer/spec#a2-class-b-transactions-also-known-as-the-multisig-method

Reply:
Thanks for the explanation!